Tests/Functional/Integration/ExtensionCreatorServiceTest.php

```diff
@@ -0,0 +1,161 @@
+
+        self::assertDirectoryExists($generatedPath);
+
+        // ✅ Check all expected files dynamically
+        foreach ($expectedFiles as $file) {
+            self::assertFileExists($generatedPath . '/' . $file, "Missing expected file: $file");
+        }
+
+        // ✅ Compare generated files with fixtures
+        $this->assertDirectoryEquals($expectedDir, $generatedPath);
+    }
+
+    public static function extensionCreationProvider(): array
+    {
+        $tempDir = sys_get_temp_dir() . '/extkickstarter_service_test';
+        $generatedPath = $tempDir . '/my_extension';
+        $extensionInfo = new ExtensionInformation(
+            'my_extension',
+            'my-vendor/my-extension',
+            'My Extension',
+            'This is a test extension',
+            '1.0.0',
+            'plugin',
+            'stable',
+            'John Doe',
+            'john@example.com',
+            'MyCompany',
+            'Vendor\\MyExtension\\',
+            $generatedPath . '/'
+        );
+
+        return [
+            'default extension' => [
+                $extensionInfo,
+                $generatedPath,
+                __DIR__ . '/Fixtures/expected_extension',
+                ['ext_emconf.php', 'README.md'], // expected files are now part of provider
+            ],
+        ];
+    }
+
+    private function deleteDirectory(string $dir): void
+    {
+        if (!is_dir($dir)) {
+            return;
+        }
+        $files = new \RecursiveIteratorIterator(
+            new \RecursiveDirectoryIterator($dir, \RecursiveDirectoryIterator::SKIP_DOTS),
+            \RecursiveIteratorIterator::CHILD_FIRST
+        );
+        foreach ($files as $file) {
+            $file->isDir() ? rmdir($file) : unlink($file);
+        }
+        rmdir($dir);
+    }
+
+    private function shouldUpdateBaseline(): bool
+    {
+        // Environment variable support
+        if (getenv('UPDATE_BASELINE') === '1') {
+            return true;
+        }
+
+        return false;
+    }
+
+    private function assertDirectoryEquals(string $expectedDir, string $actualDir): void
+    {
+        if ($this->shouldUpdateBaseline()) {
+            $this->copyDirectory($actualDir, $expectedDir);
+            self::markTestSkipped('Baseline updated: expected fixtures were overwritten with new output.');
+        }
+
+        // Normal comparison when not updating baseline
+        $expectedFiles = new \RecursiveIteratorIterator(new \RecursiveDirectoryIterator($expectedDir));
+        foreach ($expectedFiles as $file) {
+            if ($file->isDir()) {
+                continue;
+            }
+            $relativePath = str_replace($expectedDir, '', $file->getPathname());
+            $actualFile = $actualDir . $relativePath;
+
+            self::assertFileExists($actualFile, sprintf('Missing file: %s', $relativePath));
+            self::assertSame(
+                trim(file_get_contents($file->getPathname())),
```

Comment:
Just for your information. If file_get_contents fails, it returns `false` and false within `trim` will fail because of trim expects argument to be string only.